apps/api/src/app/workflows-v2/shared/schemas/delay-control.schema.ts

```diff
@@ -12,7 +12,7 @@ import {
 export const DelayTimeControlZodSchema = z
   .object({
     type: z.enum(['regular']).default('regular'),
-    amount: z.number().min(1),
+    amount: z.union([z.number().min(1), z.string()]),
```

Comment:
The `amount` field for the `digest` and `delay` steps is whether a number value or an empty string. The empty string is used in the case when the step is created to display the empty value in the field and also when the user removes the value from the field. An empty string is sent to the API for the validation.

This behavior or the number input field was just improved in this PR. I'll leave more info about it in the `AmountInput` component.

Reply:
@LetItRock, shouldn't this never happen? We autosave on blur or specific mouse and keyboard events. For delay and digest controls we shouldn't allow empty values to pass to the server.

Reply:
we will address this in DM